src/components/EmojiSuggestions.tsx

```diff
@@ -1,84 +1,64 @@
-            types: PropTypes.arrayOf(PropTypes.string.isRequired),
-        }),
-    ).isRequired,
+    emojis: SimpleEmoji[];
 
     /** Fired when the user selects an emoji */
-    onSelect: PropTypes.func.isRequired,
+    onSelect: (index: number) => void;
 
     /** Emoji prefix that follows the colon */
-    prefix: PropTypes.string.isRequired,
+    prefix: string;
 
     /** Show that we can use large emoji picker. Depending on available space
      * and whether the input is expanded, we can have a small or large emoji
      * suggester. When this value is false, the suggester will have a height of
      * 2.5 items. When this value is true, the height can be up to 5 items.  */
-    isEmojiPickerLarge: PropTypes.bool.isRequired,
+    isEmojiPickerLarge: boolean;
 
     /** Stores user's preferred skin tone */
-    preferredSkinToneIndex: PropTypes.number.isRequired,
+    preferredSkinToneIndex: number;
 
     /** Meaures the parent container's position and dimensions. */
-    measureParentContainer: PropTypes.func,
-};
-
-const defaultProps = {
-    highlightedEmojiIndex: 0,
-    measureParentContainer: () => {},
+    measureParentContainer: (callback: MeasureParentContainerCallback) => void;
 };
 
 /**
  * Create unique keys for each emoji item
- * @param {Object} item
- * @param {Number} index
- * @returns {String}
  */
-const keyExtractor = (item, index) => `${item.name}+${index}}`;
+const keyExtractor = (item: SimpleEmoji, index: number): string => `${item.name}+${index}}`;
 
-function EmojiSuggestions(props) {
+function EmojiSuggestions({emojis, onSelect, prefix, isEmojiPickerLarge, preferredSkinToneIndex, highlightedEmojiIndex = 0, measureParentContainer = () => {}}: EmojiSuggestionsProps) {
     /**
      * Render an emoji suggestion menu item component.
-     * @param {Object} item
-     * @returns {JSX.Element}
      */
-    const renderSuggestionMenuItem = (item) => {
-        const styledTextArray = getStyledTextArray(item.name, props.prefix);
+    const renderSuggestionMenuItem = (item: SimpleEmoji): ReactElement => {
+        const styledTextArray = getStyledTextArray(item.name, prefix);
 
         return (
             <View style={styles.autoCompleteSuggestionContainer}>
-                <Text style={styles.emojiSuggestionsEmoji}>{EmojiUtils.getEmojiCodeWithSkinColor(item, props.preferredSkinToneIndex)}</Text>
+                <Text style={styles.emojiSuggestionsEmoji}>{EmojiUtils.getEmojiCodeWithSkinColor(item, preferredSkinToneIndex)}</Text>
                 <Text
                     numberOfLines={2}
                     style={styles.emojiSuggestionsText}
                 >
                     :
-                    {_.map(styledTextArray, ({text, isColored}, i) => (
+                    {styledTextArray.map(({text, isColored}, i) => (
                         <Text
+                            /* eslint-disable-next-line react/no-array-index-key */
                             key={`${text}+${i}`}
```

Comment:
This isn't related to this migration and was present before, but still would be nice to fix
We shouldn't use array index in keys, elements should have stable keys when array is filtered/reordered

```suggestion
                            key={text}
```

Reply:
@eVoloshchak The only concern I have is that potentially, in a rare case, there can be two the same texts.
Maybe will be better to use `isColored` as well
```
key={`${text}+${isColored}`}
```

Reply:
> The only concern I have is that potentially, in a rare case, there can be two the same texts

There shouldn't be two identical emojis, but it's better to account for such cases, you're right. I like the `isColored` idea

Reply:
Updated!